packages/ic-response-verification/src/types/verification_result.rs

```diff
@@ -1,114 +1,62 @@
-use crate::{types::VerifiedResponse, ResponseVerificationError};
+use crate::types::VerifiedResponse;
 
 #[cfg(all(target_arch = "wasm32", feature = "js"))]
 use wasm_bindgen::prelude::*;
 
-#[cfg(all(target_arch = "wasm32", feature = "js"))]
-use crate::ResponseVerificationJsError;
-
 #[cfg(all(target_arch = "wasm32", feature = "js"))]
 #[wasm_bindgen(typescript_custom_section)]
 const VERIFICATION_RESULT: &'static str = r#"
-type VerificationResult = {
-  passed: true;
+type VerificationInfo = {
   response?: VerifiedResponse;
   verificationVersion: number;
-} | {
-  passed: false;
-  verificationVersion: number;
-  reason: ResponseVerificationError;
 }
 "#;
 
 /// Result of verifying the provided request/response pair's certification.
 #[derive(Debug)]
-pub enum VerificationResult {
-    /// Verification passed
-    Passed {
-        /// Response object including the status code, body and headers that were included in the
-        /// certification and passed verification. If verification failed then this object will be
-        /// empty.
-        response: Option<VerifiedResponse>,
-        /// The version of verification that was used to verify the response
-        verification_version: u16,
-    },
-    /// Verification failed
-    Failed {
-        /// The version of verification that was used to verify the response
-        verification_version: u16,
-        /// The reason why the verification of the response has failed
-        reason: ResponseVerificationError,
-    },
+pub struct VerificationInfo {
+    /// Response object including the status code, body and headers that were included in the
+    /// certification and passed verification. If verification failed then this object will be
+    /// empty.
+    pub response: Option<VerifiedResponse>,
```

Comment:
same here with the `Option`, ins't this throwing an exception now when verification fails?